pulp_service/pulp_service/app/authorization.py

```diff
@@ -0,0 +1,47 @@
+from contextvars import ContextVar
+from base64 import b64decode
+from binascii import Error as Base64DecodeError
+import json
+import jq
+from pulpcore.app import settings
+from pulp_service.app.models import DomainOrg
+from rest_framework.permissions import BasePermission
+
+
+org_id_var = ContextVar('org_id')
+
+class DomainBasedPermission(BasePermission):
```

Comment:
Later, we can do something similar to what we have done in pulp_container: https://github.com/pulp/pulp_container/blob/aafa1c8bd01b6708df7aef5277b517a5533fb0e3/pulp_container/app/registry_api.py#L240-L246. I do not know how the deployment would look like but I guess we can have a setting that will allow us to use either this permission class or default ones.